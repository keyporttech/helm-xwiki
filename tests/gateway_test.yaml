suite: test istio gateway
templates:
  - gateway.yaml
tests:
  - it: should set values properly of Gateway
    set:
      istio.enabled: true
    release:
      name: release
    asserts:
      - isKind:
          of: Gateway
      - isAPIVersion:
          of: networking.istio.io/v1alpha3
      - equal:
          path: metadata.name
          value: xwiki-gateway
      - matchSnapshot:
          path: spec
  - it: should properly set value of VirtualService
    set:
      service.externalPort: 80
      istio.enabled: true
    asserts:
      - isKind:
          of: VirtualService
        documentIndex: 1
      - isAPIVersion:
          of: networking.istio.io/v1alpha3
        documentIndex: 1
      - equal:
          path: metadata.name
          value: xwiki
        documentIndex: 1
      - matchSnapshot:
          path: spec
        documentIndex: 1

