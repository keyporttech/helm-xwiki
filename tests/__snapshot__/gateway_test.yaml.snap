should properly set value of VirtualService:
  1: |
    gateways:
    - xwiki-gateway
    hosts:
    - '*'
    http:
    - match:
      - uri:
          prefix: /
      route:
      - destination:
          host: RELEASE-NAME-xwiki
          port:
            number: 80
should set values properly of Gateway:
  1: |
    selector:
      istio: ingressgateway
    servers:
    - hosts:
      - '*'
      port:
        name: http
        number: 80
        protocol: HTTP
